packages/domains-table/src/domains-table/domains-table-email-indicator.tsx

```diff
@@ -1,36 +1,85 @@
-import { LoadingPlaceholder } from '@automattic/components';
 import { PartialDomainData } from '@automattic/data-stores';
+import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
-//eslint-disable-next-line no-restricted-imports
-import { useGetMailboxes } from 'calypso/data/emails/use-get-mailboxes';
+import { type as domainTypes } from '../utils/constants';
+import { getDomainType } from '../utils/get-domain-type';
+import { emailManagementEdit } from '../utils/paths';
+
+interface DomainsTableEmailIndicatorProps {
+	domain: PartialDomainData;
+	siteSlug: string;
+}
 
 export const DomainsTableEmailIndicator = ( {
 	domain,
 	siteSlug,
-}: {
-	domain: PartialDomainData;
-	siteSlug?: string;
-} ) => {
-	const { __ } = useI18n();
-	const { data: mailboxes = [], isLoading } = useGetMailboxes( domain.blog_id, {
-		retry: 2,
-	} );
-
-	if ( isLoading || ! siteSlug ) {
-		return <LoadingPlaceholder style={ { width: '50%' } } />;
+}: DomainsTableEmailIndicatorProps ) => {
+	const { __, _n } = useI18n();
+
+	const domainType = getDomainType( domain );
+
+	if ( domainType !== domainTypes.MAPPED && domainType !== domainTypes.REGISTERED ) {
+		return '-';
 	}
 
-	if ( mailboxes.length === 0 ) {
+	const googleStatus = domain.google_apps_subscription?.status;
+	const titanStatus = domain.titan_mail_subscription?.status;
+
+	let message = '';
+
+	if ( googleStatus && ! [ 'no_subscription', 'other_provider' ].includes( googleStatus ) ) {
```

Comment:
#Could you update that code to use the `hasGSuiteWithUs()` selector please? We should really avoid referring directly to those constants 🙂.